### PR TITLE
Larger channel cards in search results

### LINKED ISSUE #9703: Card mode when showing channel list item search results
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I'm aware that this is a request for NewPipe itself and that requests for adding a new service need to be made at [NewPipeExtractor](https://github.com/TeamNewPipe/NewPipeExtractor/issues).
- [X] I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Feature description
In search results, show channel card while Card mode is selected. Mock below.

Follow-up for the comment https://github.com/TeamNewPipe/NewPipe/pull/9310#issuecomment-1382733611

image

### Why do you want this feature?

Card view mode is the best candidate to display information displayed in the above mock compared to list and grid modes. A channel card shall have larger thumbnail and upto 8 lines of description.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by mahendranv**
#### What is it?

- [ ] Bugfix (user facing)
- [x] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

These changes take effect when card mode is selected for item view.

- This card is **NOT** applicable for subscriptions list fragment
  
- Thumbnail larger (100dp) than the usual (92dp) throughout the app
  
- More similar to List item with more description content
  
  - Description max line count is 8 (normally 3)
    

#### Before/After Screenshots/Screen Record

| Before: UI follows the same List mode item   |  After: New Channel card will be displayed  |
|---|---|
|image | image|

#### Fixes the following issue(s)

- Fixes #9703

- #### APK testing
  
  
  
  
  
  The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.
  

#### Due diligence

- [x] I read the [[contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md)](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mahendranv** [comment]
@Stypox Please check this small PR.

File: app/sampledata/channels.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (Stypox) [inline app/sampledata/channels.json:1]: Oh, never knew you could do this ;-)

File: app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionFragment.kt [hunk 2]
```diff
@@ -380,16 +402,13 @@ class SubscriptionFragment : BaseStateFragment<SubscriptionState>() {
...
         when (result) {
             is SubscriptionState.LoadedState -> {
                 result.subscriptions.forEach {
                     if (it is ChannelItem) {
                         it.gesturesListener = listenerChannelItem
-                        it.itemVersion = when {
-                            shouldUseGridLayout -> ChannelItem.ItemVersion.GRID
-                            else -> ChannelItem.ItemVersion.MINI
-                        }
+                        it.itemVersion = getItemVersion(itemViewMode)
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionFragment.kt:411]: what about (and also removing the `getItemVersion`, maybe move its description to `shouldUseGridForSubscription`)
```suggestion
                        it.itemVersion = if shouldUseGridForSubscription()
                            ChannelItem.ItemVersion.GRID
                        else
                            ChannelItem.ItemVersion.MINI
```

**Review by Stypox — CHANGES_REQUESTED** [review]
Thanks!

File: app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionFragment.kt (hunk 2 above)
Comment (mahendranv) [reply app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionFragment.kt]: Done

**Review by Stypox — APPROVED** [review]
I tested and everything seems to work fine. Thank you!

**Pull request merged**